pkg/storage/kubernetes.go

```diff
@@ -57,6 +58,31 @@ func NewKubernetesIPAM(containerID string, ipamConf whereaboutstypes.IPAMConfig)
 	if err != nil {
 		return nil, err
 	}
+	lockName := getNormalizedIpRangeStr(ipamConf.Range)
+	poolLock := &whereaboutsv1alpha1.IPPoolLock{
+		ObjectMeta: metav1.ObjectMeta{Name: lockName, Namespace: namespace},
+	}
+	// try creating ip pool lock, retry (if already exists) until it succeeds
+	// this makes serial access to ip pool across cluster wide and this might decrease
+	// the load considerably on k8 api server because of too many update calls, this can
+	// happen due to retries upon status conflict errors
+	for {
```

Comment:
Before calling Create(), the client can do a Get() for the resource. A successful Get() will indicate that the lock is being held by someone else and it should wait for sometime and retry.